x/oracle/keeper/slash_test.go

```diff
@@ -160,46 +160,72 @@ func TestInvalidVotesSlashing(t *testing.T) {
+	priceVoteFromVal := func(valIdx int, block int64, erate sdk.Dec) {
+		MakeAggregatePrevoteAndVote(t, input, msgServer, block,
+			types.ExchangeRateTuples{{Pair: pair, ExchangeRate: erate}},
+			valIdx)
+	}
+	input.OracleKeeper.WhitelistedPairs.Insert(input.Ctx, pair)
+	perfs := input.OracleKeeper.UpdateExchangeRates(input.Ctx)
+	require.EqualValues(t, 0, perfs.GetTotalRewardWeight())
 
-		// Account 2, govstable
-		MakeAggregatePrevoteAndVote(t, input, h, 0, types.ExchangeRateTuples{{Pair: asset.Registry.Pair(denoms.NIBI, denoms.NUSD), ExchangeRate: randomExchangeRate}}, 1)
-		// Account 3, govstable
-		MakeAggregatePrevoteAndVote(t, input, h, 0, types.ExchangeRateTuples{{Pair: asset.Registry.Pair(denoms.NIBI, denoms.NUSD), ExchangeRate: randomExchangeRate}}, 2)
+	block := input.Ctx.BlockHeight() + 50
+	priceVoteFromVal(0, block, TEST_ERATE)
+	priceVoteFromVal(1, block, TEST_ERATE)
+	input.Ctx = input.Ctx.WithBlockHeight(block + 100)
 
-		input.OracleKeeper.UpdateExchangeRates(input.Ctx)
-		// input.OracleKeeper.SlashAndResetMissCounters(input.Ctx)
-		// input.OracleKeeper.UpdateExchangeRates(input.Ctx)
-		require.Equal(t, i+1, input.OracleKeeper.MissCounters.GetOr(input.Ctx, ValAddrs[0], 0))
-	}
+	allowedMissPct := sdk.OneDec().Sub(minValidPerWindow)
+	allowedMissPeriods := allowedMissPct.MulInt64(votePeriodsPerWindow).
+		TruncateInt64()
 
-	validator := input.StakingKeeper.Validator(input.Ctx, ValAddrs[0])
-	require.Equal(t, stakingAmt, validator.GetBondedTokens())
+	// TODO
 
-	// one more miss vote will inccur Account 1 slashing
+	t.Logf("For %v blocks, valoper0 does not vote, while 1 and 2 do.", allowedMissPeriods)
+	for idxMissPeriod := uint64(0); idxMissPeriod < uint64(allowedMissPeriods); idxMissPeriod++ {
+		block := input.Ctx.BlockHeight() + 1
+		input.Ctx = input.Ctx.WithBlockHeight(block)
 
-	// Account 2, govstable
-	MakeAggregatePrevoteAndVote(t, input, h, 0, types.ExchangeRateTuples{{Pair: asset.Registry.Pair(denoms.NIBI, denoms.NUSD), ExchangeRate: randomExchangeRate}}, 1)
-	// Account 3, govstable
-	MakeAggregatePrevoteAndVote(t, input, h, 0, types.ExchangeRateTuples{{Pair: asset.Registry.Pair(denoms.NIBI, denoms.NUSD), ExchangeRate: randomExchangeRate}}, 2)
+		valIdx := 0
+		priceVoteFromVal(valIdx+1, block, TEST_ERATE)
+		priceVoteFromVal(valIdx+2, block, TEST_ERATE)
 
-	input.Ctx = input.Ctx.WithBlockHeight(votePeriodsPerWindow - 1)
-	input.OracleKeeper.UpdateExchangeRates(input.Ctx)
-	input.OracleKeeper.SlashAndResetMissCounters(input.Ctx)
+		// Valoper 0 votes outside band (miss)
+
+		perfs := input.OracleKeeper.UpdateExchangeRates(input.Ctx)
+		missCount := input.OracleKeeper.MissCounters.GetOr(input.Ctx, ValAddrs[0], 0)
+		require.EqualValues(t, 0, missCount, perfs.String())
+	}
+
+	t.Log("valoper0 should not be slashed")
+	validator := input.StakingKeeper.Validator(input.Ctx, ValAddrs[0])
+	require.Equal(t, TEST_STAKING_AMT, validator.GetBondedTokens())
+
+	// // one more miss vote will inccur Account 1 slashing
```

Comment:
Should these tests be commented out?

Reply:
Yes, meant to just delete this part since you don't get slashed for abstaining